### ISSUE TITLE
invariant_booleans false positive - decrement

### ISSUE THREAD

**Issue by nt4f04uNd** [dart-lang/linter#3257]
```dart
int i = 10;

void main() {
  if (i > 0) {
    i--;
    if (i == 0) { // < --- warning
      // do something
    }
  }
}
```

Changing `i--;` to `i -= 1;` removes false positive

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
